Source/AwsCommonRuntimeKit/io/Stream.swift

```diff
@@ -4,106 +4,99 @@
 import AwsCIo
 import Foundation
 
-// swiftlint:disable trailing_whitespace
+/// This is for streaming input.
+public protocol IStreamable {
 
-private var vtable = aws_input_stream_vtable(seek: doSeek,
-        read: doRead,
-        get_status: doGetStatus,
-        get_length: doGetLength,
-        acquire: { _ = Unmanaged<AwsInputStream>.fromOpaque($0!.pointee.impl).retain() },
-        release: { Unmanaged<AwsInputStream>.fromOpaque($0!.pointee.impl).release() })
+    /// Optional, throws length not supported error by default
+    func length() throws -> UInt64
 
-public class AwsInputStream {
-    var rawValue: aws_input_stream
-    let awsStream: AwsStream
-    public var length: Int64
-    public init(_ impl: AwsStream, allocator: Allocator = defaultAllocator) {
-        length = Int64(impl.length)
-        awsStream = impl
-        rawValue = aws_input_stream()
-        rawValue.vtable = UnsafePointer<aws_input_stream_vtable>(&vtable)
-        rawValue.impl = Unmanaged<AwsInputStream>.passUnretained(self).toOpaque()
-    }
-}
-
-public protocol AwsStream {
-    var status: aws_stream_status { get }
-    var length: UInt { get }
+    /// (Optional) Provides an implementation to seek the stream to a particular offset.
+    ///
+    /// - Parameters:
+    ///   - offset: The value to set the seek position.
+    ///             If the streamSeekType is .begin, offset value would be positive.
+    ///             If the streamSeekType is .end, offset value would be negative.
+    ///   - streamSeekType: The direction to seek the stream from
+    /// - Throws: Throws Seek not supported error by default.
+    func seek(offset: Int64, streamSeekType: StreamSeekType) throws
 
-    func seek(offset: Int64, basis: aws_stream_seek_basis) -> Bool
-    func read(buffer: inout aws_byte_buf) -> Bool
+    /// Read up to buffer.count bytes into the buffer.
+    /// buffer.count must not be modified.
+    /// Return the number of bytes copied.
+    /// Return nil if the end of file has been reached.
+    /// Return 0 if data is not yet available.
```

Comment:
Nit: Proper doc comment formatting and slight changes to the docs (feel free to ignore or modify)
 
```suggestion
    /// Reads up to `buffer.count` bytes and returns the number of bytes read.
    /// buffer.count must not be modified.
    ///
    /// - Parameters:
    ///     - buffer: The buffer to read from
    /// - Returns: `nil` if the end of the file has been reached. Returns `0` if the data is not yet available. Otherwise returns the number of bytes read.
```

Reply:
Thanks, fixed.